vllm/core/block/prefix_caching_block.py

```diff
@@ -133,20 +133,21 @@ def allocate_mutable(self, prev_block: Block) -> Block:
         assert_prefix_caching_block_or_none(prev_block)
 
         try:
-            return self._hashless_allocator.allocate_mutable(
+            block = self._hashless_allocator.allocate_mutable(
                 prev_block=prev_block)
+            self._blocks[block.block_id] = block
+            return block
         except BlockAllocator.NoFreeBlocksError:
             # We must check the unused cached blocks before raising OOM.
             pass
 
-        if self._unused_cached_blocks:
-            # TODO policy for selecting block to remove
-            content_hash_to_evict = next(iter(self._unused_cached_blocks))
+        if self.evictor.num_blocks > 0:
+            content_hash_to_evict = self.evictor.evict()
 
             # Clear content hash mapping; the block will be overwritten.
-            del self._cached_blocks[content_hash_to_evict]
+            del self._blocks[self._cached_blocks[content_hash_to_evict]]
+            block_id = self._cached_blocks.pop(content_hash_to_evict)
```

Comment:
nit: reverse the order to require only 1 hash lookup instead of 2

Reply:
btw we will need to profile the BlockManagerV2. my understanding from https://github.com/vllm-project/vllm/pull/4270 is that the scheduler is pretty sensitive to µs overhead

Reply:
yeah if the function is called for each iteration https://github.com/vllm-project/vllm/blob/c1b4e4157c0b4154f950adaea85a259fc629c758/vllm/core/scheduler.py#L429, having additional 5+us overhead for the function slows down the performance

Reply:
Got it, it make sense, would apply this change